src/Model/BufferRenderer.re

```diff
@@ -14,6 +14,7 @@
 type terminal = {
   title: string,
   id: int,
+  insertMode: bool,
```

Comment:
Deciding where to manage this state was a bit tricky (whether the terminal is in insert mode or not).

A few possibilities:
- We could store on the `Buffer`, but the buffer doesn't know about terminals...
- We could store it on the `Editor`, but the editor doesn't know about terminals...

For now, I put this in the `Terminal` buffer renderer state - when it is in insert mode, we use the terminal renderer. When it is in normal mode, we use the editor surface renderer. We also have to orchestrate with the buffer management in Vim to ensure the buffer is populated with the latest lines - we handle this in the transition from insert mode -> normal mode.

Reply:
Seems to me like a good place to put it. Are there any issues with having it here?

Reply:
Nope! I think this is the best place, for now. It just goes back to some of the issues we discussed about the buffer renderers in general (ie, they exist as an independent entity of buffers and editors - so I think those problems and caveats apply).